content/blog/migrating-vsphere-virtual-volumes-to-kubevirt-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,432 @@
+
+Create the PVC and attach the PVC as a disk to the running KubeVirt VM.
+
+```shell
+$ kubectl create -f pvc-clone.yaml
+```
+
+Next, login to OKD.
+
+![](/img/attach-disk.png "Attach disk to VM")
+
+Once attached, a “migrate” playbook will mount the disk in inside the VM and fire up the database and test it.
+
+```yaml
+---
+# With PVC attached to the KubeVirt VM, mount and start MariaDB
+- hosts: kubevirt
+  become: yes
+  tasks:
+  - name: Mount filesystem and persist
+    mount:
+      path: "{{ var_lib_mysql }}"
+      state: mounted
+      src: "LABEL={{ filesystem_label }}"
+      fstype: ext4
+
+  - name: Ensure correct perms and SELinux labels
+    file:
+      path: "{{ var_lib_mysql }}"
+      owner: mysql
+      group: mysql
+      mode: "0755"
+      seuser: system_u
+      serole: object_r
+      setype: mysqld_db_t
+      selevel: s0 
+      recurse: yes
+  - name: Enable MariaDB
+    service:
+      name: mariadb
+      state: started
+      enabled: true
+
+  - name: Test Test_DB
+    shell: mysql -t < test_employees_sha.sql && mysql -t <<< 'select @@hostname as 'VM';'
+    args:
+      chdir: "{{ ansible_user_dir }}/test_db"
+    register: test_db_results
+
+  - name: Show results
+    pause:
+      prompt: "{{ test_db_results.stdout }}"
+      seconds: 10
+```
+
+Executing the playbook should reveal the test results and we can visually inspect the database is actually running on the KubeVirt VM.
+
+```shell
+...
++---------+--------+
+| summary | result |
++---------+--------+
+| CRC     | OK     |
+| count   | OK     |
++---------+--------+
++------------------+
+| VM               |
++------------------+
+| mariadb-kubevirt |
++------------------+:
+ok: [mariadb-kubevirt]
+
+PLAY RECAP *************************************************************************
+mariadb-kubevirt           : ok=6    changed=4    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
+```
+
+Success! The cloning process worked. Now, before finalizing with the actual volume. Let’s run a “detach” playbook to stop the database and unmount the filesystem.
+
+```yaml
+---
+# With PVC attached to the KubeVirt VM, stop MariaDB and unmount the filesystem
+- hosts: kubevirt
+  become: yes
+  tasks:
+  - name: Make sure MariaDB is stopped and disabled
+    service:
+      name: mariadb
+      state: stopped
+      enabled: no
+
+  - name: Remove data disk mounts
+    mount:
+      path: "{{ var_lib_mysql }}"
+      state: absent
+```
+
+In the OKD UI, detach the volume and delete the PVC.
+
+![](/img/detach-disk.png "Detach disk from VM")
+
+```shell
+$ kubectl delete -f pvc-clone.yaml
+```
+
+Next, let’s migrate the workload permanently.
+
+# Finalizing
+
+The last step involves simply shutting down the source vSphere VM to ensure the source volume isn’t accessed. Once the VM is properly shutdown. Create a PVC with the “importVolumeName” annotation to move the volume out of the vVol datastore and make it into a PV.
```

Comment:
The second sentence is incomplete. I think you meant it to be a part of the sentence that follows:

"Once the VM is properly shut down, create a PVC with..."  (shut down should also be two words)